### ISSUE TITLE
No rule to make target '.deps/nicinfo_xdr.Plo' when compiling open-vm-tools 10.0.5 on FreeBSD 10.2-RELEASE

### ISSUE THREAD

**Issue by douglas-carmichael** [vmware/open-vm-tools#52]
When I tried to configure open-vm-tools 10.0.5 on FreeBSD 10.2-RELEASE, the configure script executed properly, but I was unable to build:

```
$ gmake
Making all in lib
gmake[1]: Entering directory '/usr/home/dcarmich/open-vm-tools-10.0.5-3227872/lib'
Making all in guestRpc
gmake[2]: Entering directory '/usr/home/dcarmich/open-vm-tools-10.0.5-3227872/lib/guestRpc'
Makefile:331: .deps/nicinfo_xdr.Plo: No such file or directory
gmake[2]: *** No rule to make target '.deps/nicinfo_xdr.Plo'.  Stop.
gmake[2]: Leaving directory '/usr/home/dcarmich/open-vm-tools-10.0.5-3227872/lib/guestRpc'
Makefile:330: recipe for target 'all-recursive' failed
gmake[1]: *** [all-recursive] Error 1
gmake[1]: Leaving directory '/usr/home/dcarmich/open-vm-tools-10.0.5-3227872/lib'
Makefile:361: recipe for target 'all-recursive' failed
gmake: *** [all-recursive] Error 1
```

I have gmake version 4.1 installed, would this be a version issue?

**Comment by oliverkurth** [comment]
Thanks for reporting.

Can you post the arguments that you gave to the configure script, so we can try to reproduce it?

Thanks,
Oliver

**Comment by douglas-carmichael** [comment]
These are the arguments I tried.

You can take a look at the [SourceForge Issue #186](http://sourceforge.net/p/open-vm-tools/tracker/186/) for the whole issue.

``````
./configure --prefix=/usr/local --without-xmlsecurity --without-icu --disable-vgauth --disable-grabbitmqproxy --disable-deploypkg --without-x --without-dnet
``````

Also, I have patches attached to that issue that made version 9.10.2 compile with clang on FreeBSD. Since clang is the default FreeBSD compiler for 10.x and beyond, would there be any way my patches could be incorporated into the mainline open-vm-tools 10.x release?

**Comment by oliverkurth** [comment]
Yes, we are working on making it build with clang, I hope we'll have it for the next release.

Thanks again!

**Comment by ravindravmw** [comment]
Douglas, Thanks for your patch. VMware's legal processes require us to have you sign our contributor agreement to accept your patches. Could you please go through http://open-vm-tools.sourceforge.net/contribute.php? Thanks!

**Comment by stanguturi** [comment]
It's been 5 years since this bug was logged. Many things have changed.

Please check the following commits:

* lib/hashMap: Remove unused HasMap_Store() and _Retrieve() functions.
   ec88ceddf4df201a75e026ed70acd34a7ffbc9a0
* open-vm-tools: Fixes to build with FreeBSD
   5114233485a7d1da81b265515c584dfce21a056d
* Fix unused function errors thrown by clang 3.4
   256d7b2ef23dcb8b5eda287f353a877b2bea17ce

Thanks to @oliverkurth There should not be any issues building the latest open-vm-tools bundle in FreeBSD. Closing this issue. Thanks.
